README.md

```diff
@@ -2575,7 +2575,43 @@ Let's create two event handlers, a `MouseEvent` handler that will prevent the de
 
 ## TypeScript and React Hooks
 
-> Intro
+In this section, we wil cover how to use the React Hooks implemented in v16.8 in TypeScript. It will be assumed that the reader has at least basic knowledge of Hooks and generics. Generics are **heavily** used alongside Hooks, mostly to set the type(s) of the variables that will be returned from the hooks, e.g., a state variable returned from `useState`. In general, it is only necessary to type a hook if it returns a value, here' an example with `useState`:
+
+Given `useState` , we can specify the type of the returning value by declaring it in TypeScript as follows:
+
+```tsx
+  const [state, setState] = useHook<T>;
+```
+
+The variable `state`, will be of type `T`. If we defined `T` as:
+
+```tsx
+  type T = number | null;
+```
+
+Then `state` and `setState` will be defined as:
+
+```tsx
+  let state: number | null;
+  let setState: (value: number | null) => void;
```

Comment:
why `let` for the `setState` callable and not `const`?

Reply:
Oh yeah, to tell you the truth, I have no idea why I used `let` 😂
Will change it for `const` in the next commit.